src/com/danick/e2/renderer/Graphic.java

```diff
@@ -375,27 +376,23 @@ public void fillPoly(int[][] points2, Color c, int X, int Y) {
 		  }
 		  
 		  for (int x = 0; x < w-0; x++) {

```

Comment:
the local x variable has the same name as the parameter, they should be different